iron-fit-behavior.html

```diff
@@ -253,11 +253,13 @@
       if (typeof this._isRTL === 'undefined') {
         this._isRTL = window.getComputedStyle(this).direction == 'rtl';
       }
-          
       this.positionTarget = this.positionTarget || this._defaultPositionTarget;
       if (this.autoFitOnAttach) {
-        if (window.getComputedStyle(this).display === 'none') {
-          setTimeout(function() {
+        // If invisible or in a shadow root that's still not connected, wait a tick.
+        if (window.getComputedStyle(this).display === 'none' ||
+          document.body.contains(this) === false) {
```

Comment:
I saw that the element wasn't centered correctly on firefox. After debugging i realized that it had all 0s in its `getClientBoundingRect` on `attached`. I should debug this more and discover where's the issue exactly - I was under the impression that `attached` would be called in cases like this
```js
var ce = document.createElement('custom-element');
document.createElement('div').appendChild(ce);
// div not in body, though custom-element is attached!